### PR TITLE
Refactor render of SR-IOV Device plugin

### PR DESCRIPTION
**Pull request by e0ne**
This PR fixes device plugin logic to handle changes on SriovOperatorConfig including `useCDI` flag.

### CODE REVIEW

File: controllers/helper.go [hunk 1]
```diff
@@ -61,3 +77,156 @@ func GetDefaultNodeSelector() map[string]string {
 	return map[string]string{"node-role.kubernetes.io/worker": "",
 		"kubernetes.io/os": "linux"}
 }
+
+func SyncPluginDaemonObjs(ctx context.Context, client k8sclient.Client, scheme *runtime.Scheme, dp *sriovnetworkv1.SriovNetworkNodePolicy, pl *sriovnetworkv1.SriovNetworkNodePolicyList) error {
```
Comment (adrianchiris) [inline controllers/helper.go:81]: This needs to be public ?

File: controllers/sriovoperatorconfig_controller.go [hunk 2]
```diff
@@ -106,6 +106,44 @@ func (r *SriovOperatorConfigReconciler) Reconcile(ctx context.Context, req ctrl.
 		return reconcile.Result{}, err
 	}
 
+	// Fetch the SriovNetworkNodePolicyList
+	policyList := &sriovnetworkv1.SriovNetworkNodePolicyList{}
```
Comment (adrianchiris) [inline controllers/sriovoperatorconfig_controller.go:110]: The creation of policy is handled by sriovnetworknodepolicy controller i think we can rely on that controller to create the policy.

perhaps we need to fail if policy is not found in sriovnetworknodepolicy controller Reconcile call

**Comment by SchSeba** [comment]
Hi @e0ne can you please rebase this PR

**Comment by e0ne** [comment]
> Hi @e0ne can you please rebase this PR

done

File: controllers/helper.go (hunk 1 above)
Comment (e0ne) [reply controllers/helper.go:82]: no, let's make it private

File: controllers/sriovoperatorconfig_controller.go (hunk 2 above)
Comment (e0ne) [reply controllers/sriovoperatorconfig_controller.go:113]: we do it at line ##120

Comment (adrianchiris) [reply controllers/sriovoperatorconfig_controller.go:113]: returning no error and empty reconcile will not requeue, we should requeue here imo to let node policy controller create the default policy.

File: controllers/sriovoperatorconfig_controller.go [hunk 3]
```diff
@@ -109,6 +109,44 @@ func (r *SriovOperatorConfigReconciler) Reconcile(ctx context.Context, req ctrl.
...
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	defaultPolicy := &sriovnetworkv1.SriovNetworkNodePolicy{}
+	err = r.Get(ctx, types.NamespacedName{Name: consts.DefaultPolicyName, Namespace: namespace}, defaultPolicy)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			// Default policy object not found, create it.
```
Comment (adrianchiris) [inline controllers/sriovoperatorconfig_controller.go:130]: do we want to create the policy here ? shouldnt the node policy controller create it ?

Comment (e0ne) [reply controllers/sriovoperatorconfig_controller.go]: fixed

**Review by adrianchiris — APPROVED** [review]
LGTM

File: controllers/helper.go [hunk 4]
```diff
@@ -62,3 +78,156 @@ func GetDefaultNodeSelector() map[string]string {
 	return map[string]string{"node-role.kubernetes.io/worker": "",
 		"kubernetes.io/os": "linux"}
 }
+
+func syncPluginDaemonObjs(ctx context.Context, client k8sclient.Client, scheme *runtime.Scheme, dp *sriovnetworkv1.SriovNetworkNodePolicy, pl *sriovnetworkv1.SriovNetworkNodePolicyList) error {
+	logger := log.Log.WithName("syncPluginDaemonObjs")
+	logger.Info("Start to sync sriov daemons objects")
```
Comment (zeeke) [inline controllers/helper.go:84]: please, reduce verbosity
```suggestion
	logger.V(1).Info("Start to sync sriov daemons objects")
```

also on L155 and L184

**Review by zeeke — COMMENTED** [review]
Left some comments about logging verbosity. Also, I think failing unit tests relate to these changes.

File: controllers/helper.go (hunk 4 above)
Comment (e0ne) [reply controllers/helper.go]: done

File: controllers/sriovoperatorconfig_controller_test.go [hunk 5]
```diff
@@ -11,11 +11,29 @@ import (
...
+	BeforeEach(func() {
+		defaultPolicy := &sriovnetworkv1.SriovNetworkNodePolicy{}
+		defaultPolicy.SetNamespace(namespace)
+		defaultPolicy.SetName(constants.DefaultPolicyName)
+		defaultPolicy.Spec = sriovnetworkv1.SriovNetworkNodePolicySpec{
+			NumVfs:       0,
+			NodeSelector: make(map[string]string),
+			NicSelector:  sriovnetworkv1.SriovNetworkNicSelector{},
+		}
+		Expect(k8sClient.Create(goctx.TODO(), defaultPolicy)).Should(Succeed())
```
[the remark this answers is not in the document]
Comment (e0ne) [reply controllers/sriovoperatorconfig_controller_test.go:29]: I prefer such approach to make test independent of the test case configuration

**Pull request merged**